compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -37,37 +37,20 @@ namespace
 {
 
 /**
- * @brief  clipAndAverage does the following things
- *         (1) Discard the top/bottom clip_ratio (%) values from vector
- *             Ex: top_clip_ratio = 5, bottom_clip_ratio = 3
- *                 discard top 5 % values and bottom 3 % values from vector
- *         (2) Calculate the average of remaining values
+ * @brief  getMovingAverage calculates the weighted moving average of input vector
+ *         The initial value is the first element of the vector
  */
-float clipAndAverage(std::vector<float> &vector, float top_clip_ratio, float bottom_clip_ratio)
+float getMovingAverage(std::vector<float> &vector, float alpha = 0.9)
 {
-  if (bottom_clip_ratio < 0 || bottom_clip_ratio >= 100)
-    throw std::runtime_error(
-        "Clip ratio (bottom) must be greater than or equal to 0 and less than 100");
+  assert(!vector.empty());
+  assert(alpha >= 0.0 && alpha <= 1.0);
 
-  if (top_clip_ratio < 0 || top_clip_ratio >= 100)
-    throw std::runtime_error(
-        "Clip ratio (top) must be greater than or equal to 0 and less than 100");
-
-  int clip_top = std::floor(vector.size() * top_clip_ratio / 100.0);
-  int clip_bottom = std::floor(vector.size() * bottom_clip_ratio / 100.0);
-  int clipped_items = clip_top + clip_bottom;
-
-  if (clipped_items >= vector.size())
-    throw std::runtime_error("The number of clipped items must be less than that of recorded data");
-
-  // Sort
-  std::sort(vector.begin(), vector.end());
-
-  // Clip and Average
-  double res = std::accumulate(vector.begin() + clip_bottom, vector.end() - clip_top, 0.0) /
-               (vector.size() - clipped_items);
-
-  return static_cast<float>(res);
+  float curr_avg = vector[0];
+  for (int i = 1; i < vector.size(); i++)
```

Comment:
can you please use same type of `vector.size()` instead of `int` ?

Reply:
I've changed the type. Thanks.